packages/interfaces/src/receiver.ts

```diff
@@ -1,13 +1,19 @@
 import type { IDecodedMessage, IDecoder } from "./message.js";
+import type {
+  ContentTopic,
+  IAsyncIterator,
+  PubSubTopic,
+  Unsubscribe,
+} from "./misc.js";
 import type { Callback, ProtocolOptions } from "./protocols.js";
 
-type Unsubscribe = () => void | Promise<void>;
-type PubSubTopic = string;
-type ContentTopic = string;
-
 export type ActiveSubscriptions = Map<PubSubTopic, ContentTopic[]>;
 
 export interface IReceiver {
+  toAsyncIterator: <T extends IDecodedMessage>(
```

Comment:
I am not sure this is the best naming because it does not express the fact that `toAsyncIterator` is like `subscribe` except that instead of accepting a callback, it returns items in a form an async iterator.

Hence, I think it should have `subscribe` in the name. Maybe `subscribeToAsyncIterator` ?

Reply:
How about `toSubscptionIterator`?

Reply:
named to `toSubscriptionIterator` for now, will update once we agree on better naming